optimizer-fix-control: add doc for the new optimizer fix controls introduced in v7.3.0 (#14257)
pingcap/docs: optimizer-fix-controls.md

- Default value: `OFF`
- Possible values: `ON`, `OFF`
- For filters such as `c = 10 and (a = 'xx' or (a = 'kk' and b = 1))`, this variable controls whether to try to build more comprehensive scan ranges for `IndexRangeScan`.

### [`44823`](https://github.com/pingcap/tidb/issues/44823) <span class="version-mark">New in v7.3.0</span>

- Default value: `200`
- Possible values: `[0, 2147483647]`

> - To save memory, Plan Cache will not cache queries with parameters exceeding the specified number.
> - This variable controls this threshold. `0` means no limit.

Reviewer: ```suggestion
- To save memory, Plan Cache does not cache queries with parameters exceeding the specified number of this variable. `0` means no limit.
```

> - For queries with more than `200` parameters, Plan Cache will not cache their execution plans to save memory.
> - This variable controls this threshold. `0` means no limit.

Reviewer 2: ```suggestion
- To save memory, Plan Cache will not cache queries with parameters exceeding the specified number.
- This variable controls this threshold. `0` means no limit.
```

### [`44830`](https://github.com/pingcap/tidb/issues/44830) <span class="version-mark">New in v7.3.0</span>

- Default value: `OFF`

> - Possible values: `ON`、`OFF`

Reviewer: ```suggestion
- Possible values: `ON`, `OFF`
```

> - This variable controls whether Plan Cache is allowed to cache the execution plan with `PointGet` that is generated during physical optimization.

Reviewer: ```suggestion
- This variable controls whether Plan Cache is allowed to cache execution plans that use the `PointGet` operator and are generated during physical optimization.
```

Author: The main point here is whether the `PointGet` is generated during physical optimization, not the execution plan.

Reviewer: 中文里是 “此开关控制是否让 Plan Cache 对在物理优化阶段形成的 `PointGet` 计划进行缓存。” 呢？要修改下中文吗

Author: 感觉改不改都行😆 。上面说的主要的点是 PointGet 在物理优化阶段生成，不是整个执行计划。
中文说的确实是"物理优化阶段形成的 PointGet 计划”，可能更准确的表述是“如果一个执行计划包含在物理优化阶段形成的 PointGet 算子，是否让 Plan Cache 对其进行缓存”，不过现在的表述也没啥大的问题。

Reviewer: 重新修改了下英文，PTAL

### [`44855`](https://github.com/pingcap/tidb/issues/44855) <span class="version-mark">New in v7.3.0</span>

- Default value: `OFF`

> - Possible values: `ON`、`OFF`

Reviewer: ```suggestion
- Possible values: `ON`, `OFF`
```

- In some scenarios, when the `Probe` side of the `IndexJoin` contains `Selection`, TiDB overestimates the row count of the `IndexScan` severely, which makes TiDB choose other execution plans when `IndexJoin` is better.

> - TiDB has introduced code to mitigate this problem. However, it's not enabled by default because of the potential risk of regressions in execution plans.

Reviewer 2: ```suggestion
- TiDB has introduced logic to mitigate this problem. However, the logic is not enabled by default due to the potential risk of regressions in execution plans.
```

> - Default value: `OFF`
> - Possible values: `ON`、`OFF`
> - In some scenarios, when the `Probe` side of the `IndexJoin` contains `Selection`, TiDB overestimates the row count of the `IndexScan` severely, which makes TiDB choose other execution plans when `IndexJoin` is better.
> - TiDB has introduced logic to mitigate this problem. However, the logic is not enabled by default due to the potential risk of regressions in execution plans.
> - This switch controls whether to enable this mitigation logic.

Reviewer 2: ```suggestion
- Default value: `OFF`
- Possible values: `ON`、`OFF`
- In some scenarios, when the `Probe` side of an `IndexJoin` operator contains a `Selection` operator, TiDB severely overestimates the row count of `IndexScan`. This might cause suboptimal query plans to be selected instead of `IndexJoin`.
- To mitigate this issue, TiDB has introduced an improvement. However, due to potential query plan fallback risks, this improvement is disabled by default.
- This variable controls whether to enable the preceding improvement.
```
